app/src/main/java/com/github/dedis/student20_pop/PoPApplication.java

```diff
@@ -375,11 +328,48 @@ public void run() {
         });
     }
 
+    /**
+     * Handle received data messages inorder
+     *
+     * @param dataMessages List of received messages
+     */
+    public void handleDataMessages(List<Data> dataMessages) {
```

Comment:
I think it's better to have public methods before private ones, this way we are consistent

Reply:
Done